src/core/client/admin/App/GlobalNotification/useNotification.ts

```diff
@@ -0,0 +1,31 @@
+import { ReactNode, useContext } from "react";
+import { NotificationContext } from "./GlobalNotificationContext";
+
+function useNotification() {
+  const context = useContext(NotificationContext);
+  if (!context) {
+    throw new Error(
+      "useNotification must be used within a NotificationProvider"
+    );
+  }
+  const [state, dispatch] = context;
+
+  const setMessage = (message: ReactNode, timeout?: number) => {
+    dispatch({ type: "SET_MESSAGE", message });
+    if (timeout) {
+      setTimeout(() => {
+        dispatch({ type: "CLEAR_MESSAGE" });
+      }, timeout);
+    }
```

Comment:
I think the timeout needs to be cleared when the component unmounts.

Reply:
This can bite you at somepoint with ugly React warnings especially in tests.

You can use the following to know it's being unmounted.
```ts
useEffect(() => {
  return () => {
    // unmounting.
  }
}, [])
```

Reply:
I thought that was just if you were using a timer inside a `useEffect` hook?

Reply:
In this case it's unrelated. Whenever a component unmounts, state changes should no longer be triggered. In this case, the notification component is always there, so you don't see warnings. But you might run into issues in tests or when the assumption that it's always there changes.

Reply:
`useRef` hook is useful here to keep a reference of the timeout handle to clear it. https://reactjs.org/docs/hooks-reference.html#useref.